lib/npm-check-updates.js

```diff
@@ -409,7 +408,7 @@ async function run(options={}) {
 
     let pendingAnalysis = options.global ?
         analyzeGlobalPackages(options) :
-        findPackage(options).spread(_.partial(analyzeProjectDependencies, options));
+        (_.partial(analyzeProjectDependencies, options)).apply(null, await findPackage(options));
```

Comment:
I believe this is equivalent to: `analyzeProjectDependencies(options, [pkgData, pkgFile])`, however `analyzeProjectDependencies` has the signature: `function analyzeProjectDependencies(options, pkgData, pkgFile)`.

It may not be explained well in the code, but pkgFile is optional. If pkgFile is undefined, it is because the content of the package.json is piped into ncu via stdin. The problem described in #474 is somehow related to isTTY being false (which would normally imply content on stdin) when the user is not piping anything over stdin.

Reply:
The `.apply` spreads the returned array from `findPackage` onto the _.partial'ized function and thus calls it with the correct signature.

In non-tty environments where the stdin is empty it falls back to using a package.json and so pkgFile should be set, because there's actually one used. The only reason pkgFile is not set in this case is because the [Promise.all return](https://github.com/tjunnone/npm-check-updates/blob/e1432917eec472bfbc20d0766bf1554ee2a01416/lib/npm-check-updates.js#L376) is called (with undefined pkgFile) before the [stdin promise](https://github.com/tjunnone/npm-check-updates/blob/e1432917eec472bfbc20d0766bf1554ee2a01416/lib/npm-check-updates.js#L359) resolves and thus [setting pkgFile](https://github.com/tjunnone/npm-check-updates/blob/e1432917eec472bfbc20d0766bf1554ee2a01416/lib/npm-check-updates.js#L364) too late in this case.

Reply:
Ah yes. I neglected the apply.

Aha! That's the timing issue! I will put together a new PR for that and get your feedback.